### PR TITLE
Optional improved / vibrant say formatting

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Optional vibrant chat colours.

Improves the say formatting by adding a little colour to the chat.
Mostly the same as the overhead chat colour, except changed to adapt for light/ dark mode.

Uses the same mechanics as the overhead chat:
 - If you are wearing an ID colour is based on the job's name.
 - If you do not have an ID / have an unknown job you get the Unknown colour
 - Agent IDs with changed names will have the colour of the job they are changed to.

### Dark Mode

image

### <s>Retina Burn</s> Light Mode

image

### Options:

image

image

## Why It's Good For The Game

It looks beautiful, makes the chat and game look nicer in general. Improves readability.
Includes an option to disable.

## Changelog
:cl:
tweak: Names are now coloured in chat.
add: Added in a TGchat button to disable coloured names.
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
Clown should be rainbow.

**Comment by PowerfulBacon** [comment]
> 
> 
> Clown should be rainbow.

Floor Cluwne is rainbow, not that it ever talks and the hierophant's pulses ever so slightly

**Comment by ike709** [comment]
Does it default on or off?

**Comment by ike709** [comment]
I like it.

File: code/game/say.dm [hunk 1]
```diff
@@ -50,9 +50,32 @@ GLOBAL_LIST_INIT(freqtospan, list(
...
+							wearer_say_span = "centcom"
+						if("Prisoner")
+							wearer_say_span = "prisoner"
+						else
+							wearer_say_span = "unknown"
+				if(wearer_say_span)
+					spanpart2 = "<span class='name [wearer_say_span]'>"
+			else
+				spanpart2 = "<span class='name unknown'>"
+	else if(isliving(speaker))
```
Comment (PowerfulBacon) [inline code/game/say.dm]:
```suggestion
	else if(isliving(speaker) && !radio_freq)
```

**Comment by PowerfulBacon** [comment]
Waiting for https://github.com/BeeStation/BeeStation-Hornet/pull/2771#pullrequestreview-565251336 to recompile TGUI.

**Comment by ike709** [comment]
@PowerfulBacon

**Comment by PowerfulBacon** [comment]
oh yea

**Pull request merged**